src/dataframes.jl

```diff
@@ -21,12 +21,31 @@ function Plots.extractGroupArgs(group::Symbol, df::AbstractDataFrame, args...)
     Plots.extractGroupArgs(collect(df[group]))
 end
 
+# allows the passing of expressions including DataFrame columns as symbols
+function processExpr(expr, df)
+    arg = map(expr.args) do x
+        isa(x, Expr) && return processExpr(x, df)
+        if isa(x, QuoteNode) && isa(x.value, Symbol)
+            return collect(df[x.value])
```

Comment:
I think just: `return :(collect(df[$x]))`

Also you are right about recursion... you can just add an "elseif" clause to the QuoteNode check if `arg` is an `Expr`, and `return processExpr!(x, df)`